examples/src/fofb_processing.c

```diff
@@ -1,272 +1,298 @@
 /*
- *  *  * Example for demonstrating the communication between a
- *   *   * a FOFB Processing module and a client
- *    *    */
+** Example for demonstrating the communication between a HALCS client and FOFB
+** Processing SMIO.
+**/
 
 #include <getopt.h>
-#include <czmq.h>
 #include <inttypes.h>
 #include <stdio.h>
 #include <string.h>
+
+#include <czmq.h>
 #include <halcs_client.h>
 
-#define DFLT_BIND_FOLDER            "/tmp/malamute"
-
-#define DFLT_HALCS_NUMBER           0
-#define DFLT_BOARD_NUMBER           0
-
-static struct option long_options[] =
-{
-  {"help",                no_argument,         NULL, 'h'},
-  {"brokerendp",          required_argument,   NULL, 'b'},
-  {"verbose",             no_argument,         NULL, 'v'},
-  {"halcsnumber",         required_argument,   NULL, 's'},
-  {"boardslot",           required_argument,   NULL, 'o'},
-  {"ram_write_en",        required_argument,   NULL, 'w'},
-  {"ram_data_in",         required_argument,   NULL, 'd'},
-  {"ram_addr",            required_argument,   NULL, 'a'},
+#include "hw/wb_fofb_processing_regs.h" // TODO: Is this include okay to be here?
+
+#define DFLT_BIND_FOLDER    "/tmp/malamute"
+
+static struct option long_opts[] = {
+  {"help",          no_argument,        NULL, 'h'},
+  {"board-slot",    required_argument,  NULL, 'o'},
+  {"halcs-number",  required_argument,  NULL, 'n'},
+  {"channel",       required_argument,  NULL, 'c'},
+  {"get-coeffs",    no_argument,        NULL, 'g'},
+  {"set-coeffs",    required_argument,  NULL, 's'},
+  {"broker-endp",   required_argument,  NULL, 'b'},
+  {"verbose",       no_argument,        NULL, 'v'},
   {NULL, 0, NULL, 0}
 };
 
-static const char* shortopt = "hb:vo:s:w:d:a:";
-
-void print_help (char *program_name)
-{
-  fprintf (stdout, "HALCSD FOFB Controller control utility\n"
-          "Usage: %s [options]\n"
-          "\n"
-          "  -h  --help                                    Display this usage information\n"
-          "  -b  --brokerendp <Broker endpoint>            Broker endpoint\n"
-          "  -v  --verbose                                 Verbose output\n"
-          "  -o  --boardslot <Board slot number = [1-12]>  Board slot number\n"
-          "  -s  --halcsnumber <HALCS number = [0|1]>      HALCS number\n"
-          "  -w  --ram_write_en <[Read = 0 | Write = 1]>   RAM write enable\n"
-          "  -d  --ram_data_in                             RAM data\n"
-          "  -a  --ram_addr                                RAM addr\n",
-          program_name);
-}
+static const char *short_opts = "ho:n:c:gs:b:v";
+
+struct optional_args {
+  char *broker_endp;
+  int verbose;
+};
 
-/* Our read function */
-typedef halcs_client_err_e (*halcs_func_fp) (halcs_client_t *self, char *service,
-        uint32_t chan, uint32_t *value);
+enum operation {
+  SET_COEFFS,
+  GET_COEFFS
+};
+
+enum necessary_args_idx {
+  BOARD_SLOT,
+  HALCS_NUMBER,
+  CHANNEL,
+  OP
+};
+
+#define NUM_OF_NECESSARY_ARGS 4
```

Comment:
```suggestion
enum necessary_args_idx {
  BOARD_SLOT,
  HALCS_NUMBER,
  CHANNEL,
  OP,
  NUM_OF_NECESSARY_ARGS
};
```

Assim ele se atualiza automaticamente, se vc precisar mudar algo.